core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/source/scan/text/FileScanSourceOpExec.scala

```diff
@@ -0,0 +1,68 @@
+package edu.uci.ics.texera.workflow.operators.source.scan.text
+
+import edu.uci.ics.texera.workflow.common.operators.source.SourceOperatorExecutor
+import edu.uci.ics.texera.workflow.common.tuple.Tuple
+import edu.uci.ics.texera.workflow.common.tuple.schema.AttributeTypeUtils
+import org.apache.commons.io.IOUtils.toByteArray
+
+import java.io._
+import java.nio.file.{Files, Paths}
+import java.util.zip.ZipFile
+import scala.jdk.CollectionConverters.{
+  asScalaIteratorConverter,
+  enumerationAsScalaIteratorConverter
+}
+
+class FileScanSourceOpExec private[text] (val desc: FileScanSourceOpDesc)
+    extends SourceOperatorExecutor {
+
+  @throws[IOException]
+  override def produceTexeraTuple(): Iterator[Tuple] = {
+    if (desc.unzip) {
+      val zipReader = new ZipFile(desc.filePath.get)
+      val entries =
+        zipReader.entries().asScala.filterNot(entry => entry.getName.startsWith("__MACOSX"))
```

Comment:
`__MACOSX` is an operating system-related file/folder. Ideally, the scan operators should not do filtering but scan files it is given to.

I read online about this `__MACOSX` folder being an irritating flaw of old macOS. To avoid this, is it possible to detect and filter it out in the implementation of uploading a compressed file?

Reply:
After discussion, we decided to keep this design. 
I feel it doesn't make much sense to modify user files, but we want to keep the output of the file scan operator clean.